### ISSUE TITLE
NullPointerException when rating a dataset

### ISSUE THREAD

**Issue by Rdornier** [GReD-Clermont/simple-omero-client#75]
Hello @ppouchin,

I have a strange behavior when I'm using ``getMyRating`` and ``rate`` methods at the dataset level.
If the rating was initially unset, then calling both methods from a datasetWrapper gives me the following error.

```
[ERROR]   [java.lang.NullPointerException] -- 
     at fr.igred.omero.AnnotatableWrapper.rate(AnnotatableWrapper.java:440)
     at fr.igred.omero.AnnotatableWrapper$rate$0.call(Unknown Source)
     at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
     at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:125)
     at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:148)
     at DL_Image_Transfer_Annotations.transferRatings(DL_Image_Transfer_Annotations.groovy:679)
     at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
     at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
     at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
     at java.lang.reflect.Method.invoke(Method.java:498)
     at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:107)
     at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:323)
     at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1262)
     at groovy.lang.DelegatingMetaClass.invokeMethod(DelegatingMetaClass.java:226)
     at org.codehaus.groovy.runtime.callsite.PogoMetaClassSite.callCurrent(PogoMetaClassSite.java:61)
     at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallCurrent(CallSiteArray.java:51)
     at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:171)
     at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:203)
     at DL_Image_Transfer_Annotations.doTransfer(DL_Image_Transfer_Annotations.groovy:390)
     at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
     at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
     at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
     at java.lang.reflect.Method.invoke(Method.java:498)
     at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:107)
     at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:323)
     at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1262)
     at groovy.lang.DelegatingMetaClass.invokeMethod(DelegatingMetaClass.java:226)
     at org.codehaus.groovy.runtime.callsite.PogoMetaClassSite.callCurrent(PogoMetaClassSite.java:61)
     at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallCurrent(CallSiteArray.java:51)
     at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:171)
     at DL_Image_Transfer_Annotations.run(DL_Image_Transfer_Annotations.groovy:165)
     at org.codehaus.groovy.jsr223.GroovyScriptEngineImpl.eval(GroovyScriptEngineImpl.java:317)
     at org.codehaus.groovy.jsr223.GroovyScriptEngineImpl.eval(GroovyScriptEngineImpl.java:155)
     at org.scijava.plugins.scripting.groovy.GroovyScriptLanguage$1.eval(GroovyScriptLanguage.java:97)
     at javax.script.AbstractScriptEngine.eval(AbstractScriptEngine.java:264)
     at org.scijava.script.ScriptModule.run(ScriptModule.java:173)
     at org.scijava.module.ModuleRunner.run(ModuleRunner.java:165)
     at org.scijava.module.ModuleRunner.call(ModuleRunner.java:125)
     at org.scijava.module.ModuleRunner.call(ModuleRunner.java:64)
     at org.scijava.thread.DefaultThreadService.lambda$wrap$2(DefaultThreadService.java:247)
     at java.util.concurrent.FutureTask.run(FutureTask.java:266)
     at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
     at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
     at java.lang.Thread.run(Thread.java:750)
```
Then, if I set it manually in the webclient and call again the methods, no error this thrown and I can either retrieve back the rating or set a new one.

The strange thing is that it doesn't throw any error if I'm calling these methods from an ImageWrapper, even if the rating was initially unset.

I was suspecting that ``ExceptionHandler.of().handle().get()`` returns somehow ``null`` instead of an empty list but I didn't get any error when calling ``getMapAnnotations`` or ``getTags`` methods (with similar implementation).

Rémy.

**Comment by ppouchin** [comment]
Ok. This is weird. I'm looking into it.

**Comment by ppouchin** [comment]
It seems that `IMetadataPrx::loadAnnotations` returns an empty list for images, but null for projects and datasets if there is no rating.

**Comment by Rdornier** [comment]
That could indeed explain the ``NullPointerException``

**Comment by ppouchin** [comment]
I have a small fix, I'll include it in the next update.

**Comment by Rdornier** [comment]
Perfect, Thank you !

**Comment by ppouchin** [comment]
Should be fixed in 5.16.0.
